packages/components/src/dropdown-menu/stories/index.js

```diff
@@ -1,52 +1,92 @@
-/**
- * External dependencies
- */
-import { text, boolean } from '@storybook/addon-knobs';
-
 /**
  * Internal dependencies
  */
 import DropdownMenu from '../';
+import { MenuGroup, MenuItem } from '../../';
 
 /**
  * WordPress dependencies
  */
-import { menu, arrowUp, arrowDown } from '@wordpress/icons';
+import {
+	menu,
+	arrowUp,
+	arrowDown,
+	chevronDown,
+	more,
+	trash,
+} from '@wordpress/icons';
 
 export default {
 	title: 'Components/DropdownMenu',
 	component: DropdownMenu,
+	argTypes: {
+		className: { control: { type: 'text' } },
+		children: { control: { type: null } },
+		disableOpenOnArrowDown: { control: { type: 'boolean' } },
+		icon: {
+			options: [ 'menu', 'chevronDown', 'more' ],
+			mapping: { menu, chevronDown, more },
+			control: { type: 'select' },
+		},
+		menuProps: {
+			control: { type: 'object' },
+		},
+		noIcons: { control: { type: 'boolean' } },
+		popoverProps: {
+			control: { type: 'object' },
+		},
+		text: { control: { type: 'text' } },
+	},
 	parameters: {
-		knobs: { disable: false },
+		controls: { expanded: true },
+		docs: { source: { state: 'open' } },
 	},
 };
 
-export const _default = () => {
-	const label = text( 'Label', 'Select a direction.' );
-	const firstMenuItemLabel = text( 'First Menu Item Label', 'Up' );
-	const secondMenuItemLabel = text( 'First Menu Item Label', 'Down' );
-	const toggleButtonTootip = boolean(
-		'Show tooltip on a toggle button',
-		true
-	);
+const Template = ( props ) => <DropdownMenu { ...props } />;
 
-	const controls = [
+export const Default = Template.bind( {} );
+Default.args = {
+	label: 'Select a direction.',
+	icon: menu,
+	controls: [
 		{
-			title: firstMenuItemLabel,
+			title: 'First Menu Item Label',
 			icon: arrowUp,
+			// eslint-disable-next-line no-console
+			onClick: () => console.log( 'up!' ),
 		},
 		{
-			title: secondMenuItemLabel,
+			title: 'Second Menu Item Label',
 			icon: arrowDown,
+			// eslint-disable-next-line no-console
+			onClick: () => console.log( 'down!' ),
 		},
-	];
+	],
+	toggleProps: {
+		showTooltip: true,
+	},
+};
 
-	return (
-		<DropdownMenu
-			icon={ menu }
-			label={ label }
-			controls={ controls }
-			toggleProps={ { showTooltip: toggleButtonTootip } }
-		/>
-	);
+export const WithChildren = Template.bind( {} );
+WithChildren.args = {
+	label: 'Select a direction.',
+	icon: more,
+	children: ( { onClose } ) => (
```

Comment:
Weird how this doesn't show our `children` code at all 😭 Might be worth hiding the code and adding manual snippet in the story description.

<img width="724" alt="image" src="https://user-images.githubusercontent.com/555336/212662371-5b3a8a66-19ed-4c2a-b7a2-a55fd344c25d.png">

Reply:
Added custom code snippet in https://github.com/WordPress/gutenberg/pull/47149/commits/d8b8af7ca75bf4985f36451733f0c0e9b848d540

Reply:
Ugh, manual snippets look so much nicer. I wish there wasn't such a big increase in maintenance cost.